File: levelOrder/levelOrder.cpp


#include "../include/include.hpp"

class Solution
{
public:


	vector<vector<int>> levelOrder(TreeNode *root)
	{
		vector<vector<int>> res;
		queue<TreeNode *> que;
		if (root == nullptr)
			return res;
		que.push(root);
		while (!que.empty())
		{
			int size = que.size();
			vector<int> level;
			for (int i = 0; i < size; i++)
			{
				TreeNode* node = que.front();
				que.pop();
				level.push_back(node->val);
				if (node->left != nullptr)
					que.push(node->left);
				if (node->right != nullptr)
					que.push(node->right);
			}
			res.push_back(level);
		}
		return res;
	}
};

int main()
{
	Solution ss;

	vector<int> nums = {3,9,20,-1,-1,15,7};
	TreeNode *root = createTree(nums, 0);

	print_nested_container(ss.levelOrder(root));

	return (0);
}

File: levelOrder/levelOrder.py
# Definition for a binary tree node.

from typing import Optional, List
import collections

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right


class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        res = []
        q = []
        q.append(root)

        while q:
            qlen = len(q)
            level = []
            for i in range(qlen):
                node = q.pop()
                if node:
                    level.append(node.val)
                    q.append(node.left)
                    q.append(node.right)
            if level:
                res.append(level)

        return res



root = TreeNode(0)
root.left = TreeNode(1)
root.right = TreeNode(2)
root.left.left = TreeNode(3)
root.right.left = TreeNode(4)

ss = Solution()

print(ss.levelOrder(root))